src/github_repo_request_local.py

```diff
@@ -136,182 +160,219 @@ def get_last_commit_hash(repo_dir: Path):
+                df.at[index, "last_commit_hash"] = last_commit_hash
+
+            # Count test files if not already counted
+            if row["testfilecountlocal"] == -1:
+                test_file_names = list_test_files(clone_dir, args.exclude)
+                count = len(test_file_names)
+                df.at[index, "testfilecountlocal"] = count
+
+                # Write the repository URL and each test file name to the text file
+                file.write(f"Repository URL: {repo_url}\n")  # Write the repo URL
+                file.writelines(
+                    f"{name}\n" for name in test_file_names
+                )  # Write each test file name
+                file.write("\n")  # Add a blank line for separation
+                logger.info(
+                    f"Test file names for the repo `{repo_name}`"
+                    f" has been written to '{test_file_list_path}'"
+                )
+
+            processed_count += 1
 
-        # Clone only if directory doesn't exist
-        if not clone_dir.exists():
-            try:
-                logger.info(f"Trying to clone {repo_url} into {clone_dir}")
-                subprocess.run(
-                    ["git", "clone", repo_url, str(clone_dir)],
-                    check=True,
-                    capture_output=True,
+            # Processed count increment and batch check
+            if processed_count >= BATCH_SIZE:
+                # Save the DataFrame to CSV
+                df.to_csv(updated_csv_path, index=False)
+                logger.info(
+                    f"Batch of {BATCH_SIZE} repositories processed. "
+                    f"Progress saved in {updated_csv_path}."
                 )
-                logger.info(f"Successfully cloned the repo: {repo_name}")
 
-                # Fetch the last commit hash and store it in the DataFrame
-                last_commit_hash = get_last_commit_hash(clone_dir)
-                if last_commit_hash is not None:
-                    df.at[index, "last_commit_hash"] = last_commit_hash
+                # Reset the processed_count for the next batch
+                processed_count = 0
 
-                # On successful clone, increment the processed_count
-                processed_count += 1
+    # Save any remaining changes at the end of processing
+    if processed_count > 0:
+        df.to_csv(updated_csv_path, index=False)
+        logger.info(f"Final batch processed. DataFrame saved in {updated_csv_path}.")
 
-            except subprocess.CalledProcessError as e:
-                logger.error(
-                    f"Failed to clone the repo: {repo_name}." f" " f"Exception: {e}"
-                )
-                df.at[index, "testfilecountlocal"] = -1
+        # Cleanup based on user's command-line option
+        if (
+            not args.keep_clones
+        ):  # If user did not specify --keep-clones, delete the directory
+            shutil.rmtree(clone_dir)
```

Comment:
It's probably worth logging whether the local repos were kept, or deleted, so that someone reading the log knows what happened.

Reply:
Added a `logger.info` before deleting the directory